app/search_builders/hyrax/dashboard/nested_collections_search_builder.rb

```diff
@@ -26,11 +26,18 @@ def with_pagination(solr_parameters)
 
       def show_only_other_collections_of_the_same_collection_type(solr_parameters)
         solr_parameters[:fq] ||= []
-        solr_parameters[:fq] += [
-          "-" + Hyrax::SolrQueryBuilderService.construct_query_for_ids([limit_ids]),
-          Hyrax::SolrQueryBuilderService.construct_query(Hyrax.config.collection_type_index_field => @collection.collection_type_gid)
-        ]
-        solr_parameters[:fq] += limit_clause if limit_clause # add limits to prevent illegal nesting arrangements
+        if Hyrax.config.use_solr_graph_for_collection_nesting
+          solr_parameters[:fq] += [
+            "-{!graph from=id to=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_parent}}id:#{@collection.id}",
+            "-{!graph to=id from=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_child}}id:#{@collection.id}"
+          ]
```

Comment:
Looks like the first clause got missed.
```suggestion
          solr_parameters[:fq] += [
            Hyrax::SolrQueryBuilderService.construct_query(Hyrax.config.collection_type_index_field => @collection.collection_type_gid),
            "-{!graph from=id to=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_parent}}id:#{@collection.id}",
            "-{!graph to=id from=member_of_collection_ids_ssim#{' maxDepth=1' if @nest_direction == :as_child}}id:#{@collection.id}"
          ]
```